### PR TITLE
[CDAP-15982] add loading indicator to validate and get schema button

### PR DESCRIPTION
**Pull request by elfenheart**
JIRA: https://issues.cask.co/browse/CDAP-15982
Build: https://builds.cask.co/browse/CDAP-URUT183

### CODE REVIEW

File: cdap-ui/app/cdap/components/AbstractWidget/GetSchemaWidget/index.tsx [hunk 1]
```diff
@@ -15,27 +15,63 @@
...
+    if (validateProperties && typeof validateProperties === 'function') {
+      setLoading(true);
+      validateProperties(() => {
+        setLoading(false);
+      });
+    }
+  }
+
+  const loadingIcon = (
+    <span className={`fa fa-spin ${classes.spinner}`}>
```
Comment (itsanudeep) [inline cdap-ui/app/cdap/components/AbstractWidget/GetSchemaWidget/index.tsx:56]: use classnames?

**Review by itsanudeep — APPROVED** [review]
minor comment, otherwise LGTM 👍

File: cdap-ui/app/cdap/components/AbstractWidget/GetSchemaWidget/index.tsx (hunk 1 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/AbstractWidget/GetSchemaWidget/index.tsx:56]: classnames is only useful if there are conditional classes being applied

File: cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js [hunk 2]
```diff
@@ -469,6 +469,10 @@ class HydratorPlusPlusNodeConfigCtrl {
       } else {
         vm.propertyErrors = propertyErrors;
       }
+
+      if (callback && typeof callback === 'function') {
```
Comment (yukiej) [inline cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js:473]: @elfenheart What do we use this callback for?

Comment (itsanudeep) [reply cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js:473]: To update state in `GetSchemaWidget/index`

Comment (elfenheart) [reply cdap-ui/app/hydrator/controllers/create/partials/nodeconfig-ctrl.js:473]: to update the loading state from GetSchemaWidget

**Pull request merged**